src/ssl/support.cc

```diff
@@ -1292,23 +1220,25 @@ static X509 * readSslX509CertificatesChain(char const * certFilename,  STACK_OF(
         return NULL;
     X509 *certificate = PEM_read_bio_X509(bio.get(), NULL, NULL, NULL);
 
-    if (certificate && chain) {
+    if (certificate) {
 
         if (X509_check_issued(certificate, certificate) == X509_V_OK)
             debugs(83, 5, "Certificate is self-signed, will not be chained");
         else {
             // and add to the chain any other certificate exist in the file
             while (X509 *ca = PEM_read_bio_X509(bio.get(), NULL, NULL, NULL)) {
-                if (!sk_X509_push(chain, ca))
-                    debugs(83, DBG_IMPORTANT, "WARNING: unable to add CA certificate to cert chain");
+                Security::CertPointer newEntry;
+                newEntry.resetAndLock(ca);
```

Comment:
The PEM_read_bio_X509 return locked X509 object you should not use resetAndLock method. Simply replace the 1230 and 1231 lines with a line:
    Security::CertPointer newEntry(ca)

Reply:
done